hibernate-core/src/main/java/org/hibernate/engine/spi/BatchFetchQueue.java

```diff
@@ -352,7 +352,15 @@ else if ( !isCached( ce.getLoadedKey(), collectionPersister ) ) {
 				}
 			}
 		}
-		return keys; //we ran out of keys to try
+		return resize( keys, i ); //we ran out of keys to try
+	}
+
+	private Object[] resize(Object[] keys, int size) {
```

Comment:
What is the purpose of this method? In 5.x we didn't need this either so I am a bit surprised that we need this now.

Reply:
I think we can reuse the `Arrays.copyOf(T[] original, int newLength)` from JDK without bothering with creating a new method, if I didn't misunderstand your purpose.

Reply:
removed it